### ISSUE TITLE
Throws.Exception failing

### ISSUE THREAD

**Issue by ChrisMaddock** [nunit/nunit#1176]
````c#
    public class ThrowsException
    {
        [Test]
        public void StandardExceptionThrows()
        {
            //Test Fails
            Assert.That(() => ThrowingException(), Throws.Exception);
        }

        [Test]
        public void InnerExceptionThrows()
        {
            //Test Passes
            Assert.That(() => ThrowingException(), Throws.InnerException);
        }

        private int ThrowingException()
        {
            throw new NullReferenceException();
        }
    }
````

With the above test cases, `StandardExceptionThrows()` fails with a NullReferenceException. Is this the correct use of Throws.Exception, and therefore a bug? 

Additionally, `InnerExceptionThrows()` is passing - and in this example, the InnerException is null? Should that be the case?

**Comment by CharliePoole** [comment]
It's slightly strange but technically correct. The lambda is probably being converted to an ActualValueDelegate returning int and the calling sequence tries to get the value for the argument before calling That.

Try it like this () => { x = ThrowingException() };  which is a void lambda and looks like a TestDelegate.

**Comment by ChrisMaddock** [comment]
Hi Charlie,

Thanks, that looks like a valid explanation. Specifying a TestDelegate type passed the test i.e.
`Assert.That<TestDelegate>(() => ThrowingException(), Throws.Exception);`

For my own sake, I copied the test into nunit solution I pulled a couple of weeks ago, to check out where it was going. In this case, the original test does convert to an ActualValueDelegate, yet the test no longer throws - I guess there's been some changes since 10.0.1 which have resolved this. 

Either way looks like this closes this issue - thanks for your advice.
